raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -179,25 +190,27 @@ def filter_used_routes(transfers_pair, routes):
     return list(channelid_to_route.values())
 
 
-def get_payee_channel(channelidentifiers_to_channels, transfer_pair):
-    """ Returns the payee channel of a given transfer pair. """
+def get_payee_channel(
+        channelidentifiers_to_channels: typing.ChannelMap,
+        transfer_pair: MediationPairState,
+) -> typing.Optional[NettingChannelState]:
+    """ Returns the payee channel of a given transfer pair or None if it's not found """
     payee_channel_identifier = transfer_pair.payee_transfer.balance_proof.channel_identifier
-    assert payee_channel_identifier in channelidentifiers_to_channels
-    payee_channel = channelidentifiers_to_channels[payee_channel_identifier]
-
-    return payee_channel
+    return channelidentifiers_to_channels.get(payee_channel_identifier)
```

Comment:
note: this is the bugfix. some channels used during the mediation may be removed while the transfer is still not completely finalized